src/lib/assets/rbac.test.ts

```diff
@@ -0,0 +1,689 @@
+          regexName: "",
+          namespaces: [],
+          regexNamespaces: [],
+          labels: {},
+          annotations: {},
+          deletionTimestamp: false,
+        },
+      },
+    ],
+    hasSchedule: false,
+    name: "",
+    description: "",
+  },
+];
+
+describe("RBAC generation", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    const mockPackageJsonRBAC = {};
+
+    jest.spyOn(fs, "readFileSync").mockImplementation((path: unknown) => {
+      if (typeof path === "string" && path.includes("package.json")) {
+        return JSON.stringify({ rbac: mockPackageJsonRBAC });
+      }
+      return "{}";
+    });
+  });
+
+  it("should generate correct ClusterRole rules in scoped mode", () => {
+    const result = clusterRole("test-role", mockCapabilities, "scoped", []);
+
+    expect(result.rules).toEqual([
+      {
+        apiGroups: ["pepr.dev"],
+        resources: ["peprstores"],
+        verbs: ["create", "get", "patch", "watch"],
+      },
+      {
+        apiGroups: ["apiextensions.k8s.io"],
+        resources: ["customresourcedefinitions"],
+        verbs: ["patch", "create"],
+      },
+      {
+        apiGroups: [""],
+        resources: ["namespaces"],
+        verbs: ["watch"],
+      },
+      {
+        apiGroups: [""],
+        resources: ["configmaps"],
+        verbs: ["watch"],
+      },
+    ]);
+  });
+
+  it("should generate a ClusterRole with wildcard rules when not in scoped mode", () => {
+    const expectedWildcardRules = [
+      {
+        apiGroups: ["*"],
+        resources: ["*"],
+        verbs: ["create", "delete", "get", "list", "patch", "update", "watch"],
+      },
+    ];
+
+    const result = clusterRole("test-role", mockCapabilities, "admin", []);
+
+    expect(result.rules).toEqual(expectedWildcardRules);
+  });
+
+  it("should return an empty rules array when capabilities are empty in scoped mode", () => {
+    const result = clusterRole("test-role", [], "scoped", []);
+
+    expect(result.rules).toEqual([]);
+  });
+
+  it("should include finalize verbs if isFinalize is true in scoped mode", () => {
+    const capabilitiesWithFinalize: CapabilityExport[] = [
+      {
+        rbac: [
+          {
+            apiGroups: ["pepr.dev"],
+            resources: ["peprstores"],
+            verbs: ["patch"],
+          },
+        ],
+        bindings: [
+          {
+            kind: { group: "pepr.dev", version: "v1", kind: "peprstore", plural: "peprstores" },
+            isWatch: false,
+            isFinalize: true,
+            event: Event.Create,
+            model: {} as GenericClass,
+            filters: {
+              name: "",
+              regexName: "",
+              namespaces: [],
+              regexNamespaces: [],
+              labels: {},
+              annotations: {},
+              deletionTimestamp: false,
+            },
+          },
+        ],
+        hasSchedule: false,
+        name: "",
+        description: "",
+      },
+    ];
```

Comment:
These `CapabilityExport[]` objects seems to be re-used by a couple of tests. To be more direct about which keys were toggling for particular test cases, maybe we should have some sort of generator or a `defaultCapabilityExport` that we use as a starting point. Then, in each test case, the "Arrange" step involves setting the key(s) that are relevant to the particular test case.

Reply:
Made a separate issue to revisit using a generator or `defaultCapabilityExport`.